src/overlays/actors/ovl_En_Guest/z_en_guest.c

```diff
@@ -20,23 +34,200 @@ const ActorInit En_Guest_InitVars = {
     (ActorFunc)EnGuest_Update,
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Guest/EnGuest_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Guest/EnGuest_Destroy.s")
+static ColliderCylinderInit_Set3 colliderInit = {
+    { COLTYPE_UNK10, 0x00, 0x00, 0x39, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, 0x00, 0x00, 0x01 },
+    { 10, 60, 0, { 0, 0, 0 } },
+};
+
+static InitChainEntry initChain[] = {
+    ICHAIN_U8(unk_1F, 6, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_4C, 500, ICHAIN_STOP),
+};
+
+UNK_PTR D_80A50BA4[] = {
+    &D_060005FC,
+    &D_060006FC,
+    &D_060007FC,
+};
+
+void EnGuest_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnGuest* this = THIS;
+
+    if ((gSaveContext.infTable[7] & 0x40) != 0) {
+        Actor_Kill(&this->actor);
+    } else {
+        this->objBankIndex = Object_GetIndex(&globalCtx->objectCtx, OBJECT_OS_ANIME);
```

Comment:
`objBankIndex` is already part of the common Actor struct, and since this one is specific to object_os_anime, I suggest using a more specific name to reflect that.
```suggestion
        this->osAnimeBankIndex = Object_GetIndex(&globalCtx->objectCtx, OBJECT_OS_ANIME);
```